src/main/java/org/jmhsrobotics/frc2023/subsystems/Drivetrain.java

```diff
@@ -173,7 +184,9 @@ public void periodic() {
 		odometry.update(imu.getRotation2d(), leftFrontEncoder.getPosition(), rightFrontEncoder.getPosition());
 
 		double yaw = this.getYaw();
-
+		if (this.isInRamsete) {
+			return;
+		}
```

Comment:
Find a more Elegant way to switch between voltage Tank and curve Drive.
Current setup could cause the drivetrain to lockup if the ramsete command is not correctly configured or is interrupted weirdly.

My recommendation would be to have a Control Mode Enum defined inside of the Drivetrain subsystem.  (One for Curverature drive the other for VoltageTank). Then inside the setters for curvature drive/voltageTank update the instance variables and the current mode.
Lastly in periodic check the current mode and apply the correct motor commands. (Might be nice to split of methods for periodic curvature and periodicVoltageTank)

A similar flag may want to be used for the heading lock so that it could be toggled on and off regardless of the input control scheme. (Tank vs curve)